sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingSymmetricHashJoinExec.scala

```diff
@@ -206,12 +206,20 @@ case class StreamingSymmetricHashJoinExec(
   }
 
   private def processPartitions(
+      leftPartitionId: Int,
       leftInputIter: Iterator[InternalRow],
+      rightPartitionId: Int,
       rightInputIter: Iterator[InternalRow]): Iterator[InternalRow] = {
     if (stateInfo.isEmpty) {
       throw new IllegalStateException(s"Cannot execute join as state info was not specified\n$this")
     }
 
+    if (leftPartitionId != rightPartitionId) {
```

Comment:
Dumb question, but why accept two partition IDs here in a private method if they must be the same?

Reply:
I imagined some case where two RDDs have different number of partitions so that's to not restricting the function signature, but for StateStoreAwareZipPartitionsRDD it shouldn't. I think we can move this check to `StateStoreAwareZipPartitionsRDD.compute` and reduce parameter. Thanks!